### PR TITLE
feature/redis-lpush: Adding support for redis-lpush

### PR DESCRIPTION
**Pull request by m-denton**
This change will add support for Redis LPUSH operations. Related to Issue #820.

### CODE REVIEW

**Comment by osheroff** [comment]
would prefer this as just a configuration option to the redis producer

**Comment by m-denton** [comment]
@osheroff I am not seeing why the build has failed. Can you check into this and let me know how 9b1b58a is looking? Thanks.

**Comment by liulikun** [comment]
It was a flaky test. The code looks good!

**Comment by m-denton** [comment]
Thanks for the update @liulikun. Excited to have this implemented.

File: docs/docs/producers.md [hunk 1]
```diff
@@ -200,14 +200,15 @@ For more details on these options, you are encouraged to the read official Rabbi
...
-Set the output stream in `config.properties` by setting the `redis_pub_channel` property.
+Set the output stream in `config.properties` by setting the `redis_pub_channel` property or set the `redis_list_key` property when using redis_type = LPUSH.
 
 Other configurable properties are:
 
 - `redis_host` - defaults to **localhost**
 - `redis_port` - defaults to **6379**
 - `redis_auth` - defaults to **null**
 - `redis_database` - defaults to **0**
+- `redis_type` - defaults to **pubsub**
```
Comment (osheroff) [inline docs/docs/producers.md]: document the `lpush` option here

**Comment by osheroff** [comment]
good to go 👍 after tiny documentation cleanup

**Comment by m-denton** [comment]
@osheroff Good catch!

**Comment by osheroff** [comment]
nice work, thx for the PR

**Comment by m-denton** [comment]
Testing on the newest build, doesn't seem to be working as expected. Setting redis_type=lpush and redis_list_key=test in config_properties then starting Maxwell with producer=redis is still outputting via Redis pub/sub. Is there something that I missed?

**Comment by osheroff** [comment]
by "latest bulid" do you mean the latest release?  This PR is merged but not released.

**Comment by m-denton** [comment]
That would do it!

**Pull request merged**